components/UpdateNurse/UpdateNurse.js

```diff
@@ -0,0 +1,103 @@
+import useSWR from "swr";
+import { useRouter } from "next/router";
+import {
+  FormContainer,
+  InputGroup,
+  Label,
+  Input,
+  Select,
+  AddButtonStyled,
+  HeaderStyled,
+} from "@/components/FormAddNurse/FormAddNurse.styled";
+
+export default function UpdateNurse({ nurseData }) {
+  const { mutate } = useSWR("/api/nurses"); //check if db has new data - re-validation!
+  const router = useRouter();
+  const { id } = router.query;
```

Comment:
Same logic applies here: you want to keep your components as dumb as possible.
Move logic away from here and then pass props down.